tests/testsuite/update.rs

```diff
@@ -1936,20 +2042,25 @@ fn update_breaking_specific_packages() {
     Package::new("ws", "2.0.0").publish();
     Package::new("transitive-incompatible", "2.0.0").publish();
 
-    p.cargo("update --breaking just-foo shared ws")
-        .with_status(1)
-        .with_stderr_contains("[ERROR] unexpected argument '--breaking' found")
-        .run();
-
-    p.cargo("update just-foo shared ws")
-        .with_stderr(
+    // FIXME: This should not upgrade just-bar
```

Comment:
Tracking FIXME

Reply:
What if in the `breaking` case, we called `resolve_ws` instead of `ops::update_lockfile(&ws, &update_opts, &upgrades)?;`  That would make the behavior more like `cargo check` without having to muck with `keep`

Reply:
That works indeed. Thanks. Although I did have to pepper the code with `dry_run` arguments. Not sure which approach is best. The actual FIXME above was unrelated to this, as it was about not honouring the `to_update` collection.

Reply:
I think I'm missing something.  Could you explain the situation?

Reply:
The fix for the FIXME above was this:

```
    if !opts.to_update.is_empty() && !opts.to_update.contains(&name.to_string()) {
        trace!("skipping dependency `{}` not selected for upgrading", name);
        return Ok(dependency);
    }
```

Your suggestion to use `resolve_ws` works as well as ops::update_lockfile. The only difference seemed to be that `resolve_ws` insisted on writing the lock file, so it needed a `dry_run` argument. This caused a number of changes across the code base. If that looks fine to you, let's keep it. Otherwise, we can revert to using ops::update_lockfile.

Reply:
Honestly,. I would just pass the parameters directly and not use `opts` at all in this case (which means `UpdateOptions` doesn't need `breaking`).

The alternative is to move all of the logic in the `command/update.rs` into the `ops` and `UpdateOptions` can be passed to that but the lower level functions probably shouldn't accept it.

Reply:
Thanks!  I thought this was going in a different direction.  Unsure which approach is better overall though we can see over time.  I just hesitate over the modification of `keep` so avoiding that is a win for me.